packages/core/src/Timer.ts

```diff
@@ -0,0 +1,92 @@
+type Tick = {
+    // Result of Date API. Unix timestamp in ms.
+    timestamp: number,
+    // Result of performance.now API. Timestamp in ms (with microsecond precision)
+    // since JS context start.
+    timestamp_rn: number | null
+}
+
+const START_LABEL = "__start"
+const STOP_LABEL = "__stop"
+
+/**
+ * Simple timer which records time ticks. Shouldn't be re-used once stopped.
+ * All timestamps/durations returned are in milliseconds.
```

Comment:
why do we have this limitation?
can't we have an API like that:
```
const timer = Timer() // its constructor sets its start_time
...
const duration1 = timer.durationSinceStart() // timer doesn't change its internal state, it just returns "now() - start_time"
...
const duration2 = timer.durationSinceStart() // now we can re-use the timer without problem
```

Reply:
Timer doesn't start at the construction time, because it is constructed at the XHR context construction time, but we start time (start request) later, at the different point of time.

Regarding the limitation: what can be the need to re-use same timer again once it is stopped, considering that lifetime of this timer is bound to the lifetime of underlying request?  We could introduce `durationSinceStart` method, but then it would be several time fields in the XHR context object as before this change. I think it is better to encapsulate timing-related information in the dedicated object which is `Timer`.

Reply:
I'd rather keep it immutable and tight to the resource lifetime.

Reply:
I'm not sure how API will look like in this case. Let's suppose timer store only start time, and responsibility of keeping individual time ticks is on the caller, then `DdRumXhrContext` object will have both times and timer itself (because we have timer bound to the resource lifetime and we need to carry it around):

```
class DdRumXhrContext {
   ...
   timer,
   startTime,
   responseStartTime,
   ...
}
```
I find it a bit confusing to have timer object with start time only inside and other timestamps outside it at the same object. Is there another way to handle it if timer would store only start time?

If statement about re-use is confusing, I can simply add `.reset` method. wdyt?

Reply:
I'd say keep it this way, this is actually what I suggested above. It is better to have one instance/resource.